Source/SwiftLintFramework/Rules/Style/VerticalWhitespaceBetweenCasesRule.swift

```diff
@@ -104,6 +104,28 @@ public struct VerticalWhitespaceBetweenCasesRule: ConfigurationProviderRule {
     ]
 
     private let pattern = "([^\\n{][ \\t]*\\n)([ \\t]*(?:case[^\\n]+|default):[ \\t]*\\n)"
+
+    private func violationRanges(in file: File) -> [NSRange] {
+        return file.violatingRanges(for: pattern).filter {
+            !isFalsePositive(in: file, range: $0)
+        }
+    }
+
+    private func isFalsePositive(in file: File, range: NSRange) -> Bool {
+        // Regex incorrectly flags blank lines that contain trailing whitespace (#2538)
+        let patternRegex = regex(pattern)
+        let substring = file.contents.substring(from: range.location, length: range.length)
+        let substringRange = NSRange(location: 0, length: substring.count)
```

Comment:
this should be `substring.utf16.count` (or even better, you can use `substring.fullNSRange`)